### PR TITLE
fix(orch): reject review artifacts with unroutable finding items

### LINKED ISSUE #810: review-artifact-check accepts findings missing required schema fields (unroutable category)
## Summary

`review-artifact-check` returns `{"ok": true, "reason": "valid"}` for a review artifact whose `suggestions[]` items are missing every required field except `title` and `location`. Because the orchestrator routes on `category` (`fix` → dev agent, `issue` → audit), an artifact that passes the gate can still be unroutable, and the orchestrator only discovers this when it tries to read the fields.

## Observed

A QA agent wrote a schema-noncompliant artifact — items shaped `{title, location, detail, severity}` instead of the `review-finding` schema's `{id, title, location, description, recommendation, priority, estimate, category}`:

```json
{
  "agent": "reviewer-arch",
  "verdict": "pass",
  "blockers": [],
  "suggestions": [
    {
      "title": "Two shell-level symbol-to-display resolvers coexist",
      "location": "/abs/path/ht-app/src/instrument_link.rs:21 (instrument_name)",
      "detail": "…",
      "severity": "low"
    }
  ],
  "qa_metadata": { "arch_review": { "overall_score": 8.4, "pass": true } }
}
```

`review-artifact-check <worktree> reviewer-arch <boundary>` returned:

```json
{ "ok": true, "path": "…/review-reviewer-arch-….json", "reason": "valid" }
```

## Expected

Some signal that the finding items are unusable. The check already validates existence, `mtime >= delegated_at`, `jq -e '.verdict'`, the `qa_metadata.review_performed: false` no-review case, and — per its own documentation — that "an artifact declaring `qa_metadata` still carries its `blockers[]`/`suggestions[]` arrays (reason `incomplete` when they were lost in the write)".

That last rule shows the intent is already to catch structurally unusable artifacts, not only missing ones. An artifact whose arrays are present but whose items carry none of the required fields is the same class of defect: the reviewer's work is there in prose but cannot be consumed.

## Impact

Moderate. `review-artifact-check` is documented as "the sole review-pr completion condition", so a passing result is what the orchestrator relies on to treat a reviewer as complete. In this case the orchestrator caught it downstream only because it inspected `.category` to route, found it absent, and issued a re-delegation. A workflow that trusted `ok: true` and iterated `suggestions[]` mechanically would silently drop every finding — a `category`-less item matches neither the `fix` filter nor the `issue` filter.

The failure mode is quiet: verdict and summary are intact and correct, so the artifact looks healthy in every log line, while the findings it exists to carry are unroutable.

## Suggested fix

Extend the validator to check item shape when `blockers[]` / `suggestions[]` are non-empty, reusing the existing `incomplete` reason:

- every item has `id`, `title`, `location`, `description`, `recommendation`, `priority`, `estimate`
- every `suggestions[]` item additionally has `category` ∈ {`fix`, `issue`}
- `priority` ∈ 1..4, `estimate` ∈ 1..5

A `jq` predicate over the arrays would be enough, and it composes with the existing gate ordering (missing → invalid → incomplete → valid). This also converts a class of agent output error into a deterministic re-delegation trigger via the documented `ok == false` path in `review-pr.md` § 3.1, rather than depending on the orchestrator noticing.

## Environment

- vstack orch skill v2.0.0, reviewer skill v1.0.0
- Schema: `.agents/skills/reviewer/schemas/review-finding.md`
- Linux, `review-artifact-check` in both glob and `--file` modes (both returned `valid`)

### PR DESCRIPTION
**Pull request by bmethod**
Closes #810

## Problem

`review-artifact-check` returned `{"ok": true, "reason": "valid"}` for an artifact whose `suggestions[]` items carried none of the required `review-finding` fields — e.g. `{title, location, detail, severity}` instead of `{id, title, location, description, recommendation, priority, estimate, category}`. Because the orchestrator routes suggestions on `category` (`fix` → dev, `issue` → audit), a category-less item matches neither filter and every finding is silently dropped. The gate passed an unroutable artifact; the orchestrator only found out when it read fields that weren't there.

## Fix

A new `finding_item_detail` check, gated on the `qa_metadata` declaration for parity with the existing arrays-lost `incomplete` rule. It requires each present `blockers[]`/`suggestions[]` item to carry the required item fields from `reviewer/schemas/review-finding.md` — the seven base fields for both arrays, plus `category ∈ {fix,issue}` for suggestions (blockers correctly do **not** require category; the schema marks it "Suggestions only"). Empty arrays carry no items and stay valid.

Rejection reuses reason `incomplete` (same class: an artifact whose findings can't be consumed) and adds an **additive** `detail` field naming the first offending item and field, e.g. `"suggestions[0]: missing/invalid category, description"`. The `{ok, path, reason}` shape is unchanged, so existing callers that branch on `.ok` are unaffected.

The re-delegation message in `review-pr.md` now spells out the required fields so a reviewer that produced a malformed artifact is told exactly how to fix it.

## Why this doesn't break real reviews

Verified the repo's own reviewers comply: `reviewer/workflows/qa-review.md` § step 1 instructs agents to build JSON **per `review-finding.md`**, and that schema's canonical example emits all seven base fields plus `category` on every suggestion. The malformed `{title, location, detail, severity}` shape in the issue was an aberration from a QA agent, not the prescribed output — exactly the class this catches. A rejected artifact triggers the existing one-shot re-delegation (now with the field list), not a hard failure; local/external review is advisory.

## Docs

`review-finding.md`, `SKILL.md`, `review-pr.md`, and `submit-pr.md` updated so the contract prose matches the tightened check, including the new `detail` field.

## Verification

- `bash skills/orch/tests/run-all.sh` — all 21 files pass, including 128 lines of new coverage in `review_artifact_check.sh`: the exact malformed shape → `ok:false`; a compliant artifact → `ok:true`; empty arrays → `ok:true`; an item missing only `category` → `ok:false`; and all pre-existing cases (missing / stale / no-verdict / `review_performed:false` / arrays-lost) unchanged.

### CODE REVIEW

File: skills/orch/scripts/review-artifact-check [hunk 1]
```diff
@@ -84,6 +95,54 @@ qa_shaped_incomplete() {
...
+        as $base
+      | ( [ (.blockers?    | if type == "array" then . else [] end
+              | to_entries[] | {arr: "blockers",    i: .key, item: .value}),
+            (.suggestions? | if type == "array" then . else [] end
+              | to_entries[] | {arr: "suggestions", i: .key, item: .value}) ]
+          | map(
+              .arr as $arr | .i as $i | .item as $item
+              | ($base + (if $arr == "suggestions" then ["category"] else [] end))
+                as $req
+              | [ $req[] | select((($item | type) != "object") or ($item[.] == null)) ]
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/orch/scripts/review-artifact-check:130]: Verified against both sources and fixed in 115b8032. `review-finding.md`'s field table constrains `priority` to 1-4 and `estimate` to 1-5, and issue #810's acceptance list includes `priority ∈ 1..4, estimate ∈ 1..5` — so this was under-enforcement, not out of scope.

The item predicate now flags a present `priority`/`estimate` that is non-numeric or out of range, with a `detail` naming the field (e.g. `suggestions[0]: missing/invalid priority(not 1..4)`). Design points: a null value stays `$missing`'s concern (not double-reported), blockers carry the same numeric constraint (they require the 7 base fields, just not `category`), and boundary values (priority 1 and 4, estimate 1 and 5) stay valid — verified with an explicit test so this isn't off-by-one.

New tests cover priority:5, a string estimate, a blocker priority:0, and the 4/5 boundary. Full orch suite green (21 files).

File: skills/orch/scripts/review-artifact-check [hunk 2]
```diff
@@ -127,6 +201,11 @@ if [[ "${1:-}" == "--file" ]]; then
     emit false "$file" "incomplete"
     exit 1
   fi
+  detail="$(finding_item_detail "$file")"
+  if [[ -n "$detail" ]]; then
+    emit false "$file" "incomplete" "$detail"
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/orch/scripts/review-artifact-check:206]: Fixed in 8367ed1b. README.md:95 now documents the `incomplete` rejection for unusable findings (missing/non-array `blockers[]`/`suggestions[]`, or a present item omitting a required `review-finding` field — including `priority` 1-4, `estimate` 1-5, and `category ∈ {fix,issue}` on suggestions) and the additive `detail` field, keeping the `{ok, path, reason}` promise accurate. Checked the sibling `dev-artifact-check` paragraph at :99 — unaffected by this change.

File: skills/reviewer/schemas/review-finding.md [hunk 3]
```diff
@@ -61,7 +61,7 @@ Create the JSON artifact with the active harness file-write/edit tool. In Codex,
...
+Declaring `qa_metadata` also commits the artifact to usable findings. `review-artifact-check` rejects such an artifact with reason `incomplete` when its `blockers[]` or `suggestions[]` is missing or not an array (a truncated write can keep `verdict`/`summary` while silently losing the findings, so the verdict cannot be trusted), AND when a present `blockers[]`/`suggestions[]` item omits a required item field below — `id`, `title`, `location`, `description`, `recommendation`, `priority` (a number in 1–4), `estimate` (a number in 1–5), plus `category ∈ {fix,issue}` for suggestions (the orchestrator routes suggestions on `category`, so a category-less item is unroutable; the rejection's `detail` field names the offending item and field). `questions[]` is exempt: it is PR-comment-triage-only. Artifacts without `qa_metadata` keep the tolerant existence + `verdict` validation.
```
[the remark this answers is not in the document]
Comment (bmethod) [reply skills/reviewer/schemas/review-finding.md]: Fixed in 826af68e — reworded to "rejects such an artifact with reason `incomplete` under either of two independent conditions: ... or ...", matching the checker (missing arrays OR a malformed item independently triggers rejection).

Checked the sibling doc copies for the same ambiguity: the `review-pr.md` phrasing uses the inverse, positive framing ("carries usable findings — arrays present AND items well-formed"), where the AND is correct because usability requires both and rejection is the negation. So only this sentence needed the fix.
